src/components/DpInput/DpInput.vue

```diff
@@ -37,248 +37,252 @@
+    type: String,
+    required: false,
+    default: ''
+  },
+
+  dataDpValidateIf: {
+    type: String,
+    required: false,
+    default: ''
+  },
+
+  dataDpValidateShouldEqual: {
+    type: String,
+    required: false,
+    default: ''
+  },
+
+  disabled: {
+    type: Boolean,
+    required: false,
+    default: null
+  },
+
+  /**
+   * Use: when input field used with Icon || Button, then input field has padding right.
+   */
+  hasIcon: {
+    type: Boolean,
+    required: false,
+    default: false
+  },
+
+  id: {
+    type: String,
+    required: true
   },
 
-  watch: {
-    value: function () {
-      this.currentValue = this.value
+  label: {
+    type: Object,
+    default: () => ({
+      bold: true,
+      hide: false,
+      hint: '',
+      text: '',
+      tooltip: ''
+    }),
+    validator: (prop) => {
+      return Object.keys(prop).every(key => ['bold', 'hide', 'hint', 'text', 'tooltip'].includes(key))
     }
   },
 
-  methods: {
-    handleEnter (event) {
-      if (this.preventDefaultOnEnter === true) {
-        event.preventDefault()
-      }
-      this.$emit('enter')
+  /**
+   * Limit the maximum allowed number of characters to the given amount.
+   */
+  maxlength: length,
+
+  /**
+   * Define the minimum number of characters that need to be given.
+   */
+  minlength: length,
+
+  name: {
+    type: String,
+    required: false,
+    default: ''
+  },
+
+  pattern: {
+    type: String,
+    required: false,
+    default: ''
+  },
+
+  placeholder: {
+    type: String,
+    required: false,
+    default: ''
+  },
+
+  /**
+   * Set to false to prevent default behavior onEnter.
+   */
+  preventDefaultOnEnter: {
+    type: Boolean,
+    required: false,
+    default: true
+  },
+
+  readonly: {
+    type: Boolean,
+    required: false,
+    default: null
+  },
+
+  required: {
+    type: Boolean,
+    required: false,
+    default: null
+  },
+
+  /**
+   * When setting a number for the `size` prop, this is directly rendered
+   * as html attribute on the input element. Also, it is assumed that visual sizing
+   * based on that value shall be applied, that is why both container classes
+   * and element classes do not define any width styles when a size is set here.
+   */
+  size: {
+    type: [Number, null],
+    required: false,
+    default: null
+  },
+
+  type: {
+    type: String,
+    required: false,
+    default: 'text'
+  },
+
+  value: {
+    type: String,
+    required: false,
+    default: ''
+  },
+
+  /**
+   * Full width by default; set to 'auto' to have no width defined.
+   * @deprecated Apply width to the parent element of DpInput.
+   */
+  width: {
+    type: String,
+    default: 'w-full'
+  }
+})
+
+const emit = defineEmits(['blur', 'enter', 'focus', 'input'])
+
+const { value, maxlength, minlength, width, size, label } = toRefs(props)
+const currentValue = ref(value.value)
+
+const classes = computed(() => {
+  let cssClasses: string[] = [
+    'px-1 py-0.5 max-w-full rounded-input',
```

Comment:
Why several strings?

Reply:
for better readability (several lines). 
changed to string literal in 6bb17e6f31e49af5e6365b9e5a7cc5dfd89fcebb